### PR TITLE
feat: cors config

### PR DESCRIPTION
**Pull request by luismaximozup**
Setting default CORS settings

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package br.com.zup.bff.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.servlet.config.annotation.CorsRegistry
+
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+
+@Configuration
+class CorsConfiguration(@Value("\${web.url}") val webUrl: String) : WebMvcConfigurer {
```
Comment (luismaximozup) [inline backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt:26]:
```suggestion
class CorsConfiguration(@Value() : WebMvcConfigurer {
```

File: backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    override fun addCorsMappings(registry: CorsRegistry) {
+        registry
+                .addMapping("/**")
+                .allowedOrigins(this.webUrl)
```
Comment (luismaximozup) [inline backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt:30]:
```suggestion
```

File: backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.servlet.config.annotation.CorsRegistry
+
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+
+@Configuration
+class CorsConfiguration(@Value("\${web.url}") val webUrl: String) : WebMvcConfigurer {
+    override fun addCorsMappings(registry: CorsRegistry) {
+        registry
+                .addMapping("/**")
+                .allowedMethods("*")
```
Comment (luismaximozup) [inline backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt:30]:
```suggestion
                .allowedMethods("*")
                .allowedOrigins("*")
```

File: backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+package br.com.zup.bff.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.servlet.config.annotation.CorsRegistry
+
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+
+@Configuration
+class CorsConfiguration(@Value() : WebMvcConfigurer {
```
Comment (luismaximozup) [inline backend/src/main/kotlin/br/com/zup/bff/config/CorsConfiguration.kt:26]:
```suggestion
class CorsConfiguration : WebMvcConfigurer {
```

**Review by vitorbassozup — APPROVED** [review]

**Pull request merged**